programs/drift/src/instructions/admin.rs

```diff
@@ -4970,6 +4976,185 @@ pub fn handle_add_amm_constituent_data<'info>(
     Ok(())
 }
 
+pub fn handle_begin_lp_swap<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, LPTakerSwap<'info>>,
+    amount_in: u64,
+) -> Result<()> {
+    let ixs = ctx.accounts.instructions.as_ref();
+    let current_index = instructions::load_current_index_checked(ixs)? as usize;
+
+    let current_ix = instructions::load_instruction_at_checked(current_index, ixs)?;
+    validate!(
+        current_ix.program_id == *ctx.program_id,
+        ErrorCode::InvalidSwap,
+        "SwapBegin must be a top-level instruction (cant be cpi)"
+    )?;
+
+    // Make sure we have enough balance to do the swap
+    let constituent_in_token_account = &ctx.accounts.constituent_in_token_account;
+    validate!(
+        amount_in >= constituent_in_token_account.amount,
+        ErrorCode::InvalidSwap,
+        "trying to swap more than the balance of the constituent in token account"
+    )?;
+
+    let remaining_accounts = &mut ctx.remaining_accounts.iter().peekable();
+    let mint = get_token_mint(remaining_accounts)?;
+
+    receive(
```

Comment:
i think you need to use `send_from_program_vault` here since the CPI to token program needs to be signed by drift_signer (constituent token account owner)